ci/jenkins/mellanox/scripts/start_ci.sh

```diff
@@ -94,10 +94,18 @@ EOF
     if [[ -z "${ANTREA_CNI_PR}${ANTREA_CNI_BRANCH}" ]];then
         change_image_name $ANTREA_CNI_HARBOR_IMAGE antrea/antrea-ubuntu
     fi
-    
-    if [[ -z "$(grep hw-offload $WORKSPACE/antrea-cni/build/yamls/antrea.yml)" ]];then
-        sed -i '/start_ovs/a\        - --hw-offload' $WORKSPACE/antrea-cni/build/yamls/antrea.yml
-    fi
+
+    change_image_name antrea/antrea-ubuntu antrea/antrea-ubuntu:ci
+
+    change_k8s_resource "DaemonSet" "antrea-agent" spec.template.spec.containers[*].image "antrea/antrea-ubuntu:ci" "$WORKSPACE/antrea-cni/build/yamls/antrea.yml"
+    change_k8s_resource "DaemonSet" "antrea-agent" spec.template.spec.initContainers[*].image "antrea/antrea-ubuntu:ci" "$WORKSPACE/antrea-cni/build/yamls/antrea.yml"
+
+    change_k8s_resource "Deployment" "antrea-controller" spec.template.spec.containers[*].image "antrea/antrea-ubuntu:ci" "$WORKSPACE/antrea-cni/build/yamls/antrea.yml"
+
+    local antrea_ovs_container_index=$(get_daemonset_container_index_from_container_name "antrea-agent" "antrea-ovs" "$WORKSPACE/antrea-cni/build/yamls/antrea.yml")
+    change_k8s_resource "DaemonSet" "antrea-agent" spec.template.spec.containers["$antrea_ovs_container_index"].command[+] "--hw-offload" "$WORKSPACE/antrea-cni/build/yamls/antrea.yml"
```

Comment:
why this code instead if a single `sed` command to rename the image?
this looks a bit clunky to me, it may be a good idea to use a dedicated tool to make yaml changes, like `kustomize`. Or even update https://github.com/vmware-tanzu/antrea/blob/main/hack/generate-manifest.sh to support generating the appropriate YAML for the CI

Reply:
sed can be unreliable sometimes, for it to work in this case it needs to look for all image occurrences regardless of the value (because the name may change in the future), but in that case if a container is added with a different image, that container may fail. 

I agree with you that yaml files should be updated with a dedicated tool, and i did not know that `generate-manifest.sh` file existed, so i used the `yq` tool, which may have made things more missy, but it should be reliable.

I updated the patch to use the generate-manifest.sh to change the image, can you have a look please?